### PR TITLE
Make TemplateMatching noise level configurable

### PR DESCRIPTION
**Pull request by Routhleck**
## Summary

The per-step Gaussian noise added by `TemplateMatching` (and its
`TemplateMatching1D` / `TemplateMatching2D` subclasses) was hard-coded
to `0.1 * A * randn()` inside `get_data()`. Callers had no way to
scale that noise from outside the task.

Promote it to an explicit `noise_level` constructor argument, defaulting
to `0.1` to preserve the previous behavior. Negative values are rejected
at construction time.

## Changes

- `TemplateMatching.__init__` gains `noise_level: float = 0.1` (must be
  non-negative). `get_data()` multiplies `randn()` by `self.noise_level`
  instead of the literal `0.1`.
- `TemplateMatching1D` / `TemplateMatching2D` forward `noise_level`
  through to the base class.
- Update class docstring to reflect the new parameter.

## Tests

Add unit tests under `tests/unit/task/tracking/`:

- `test_template_matching_1d_noise_level_zero_is_clean`:
  `noise_level=0` produces a noise-free constant stimulus.
- `test_template_matching_1d_noise_level_scales_std`:
  doubling `noise_level` roughly doubles the per-pixel noise std
  (verified by comparing two runs with the same RNG seed).
- `test_template_matching_1d_invalid_noise_level_rejected`:
  negative `noise_level` is rejected with `ValueError`.
- Same two coverage tests for the 2D variant.

All 11 tests in the tracking task suite pass, and the full
`uv run pytest -m "not slow"` run is green (139 passed, 4 skipped).

## Backward compatibility

`noise_level` defaults to `0.1`, so existing callers see identical
behavior. No public API is removed or renamed.

## Summary by Sourcery

Make the per-step Gaussian noise level in template matching tracking tasks configurable and validate its value.

New Features:
- Add a configurable noise_level parameter to TemplateMatching and its 1D/2D variants to control per-step Gaussian noise strength.

Bug Fixes:
- Reject negative noise_level values at construction time to prevent invalid noise configurations.

Enhancements:
- Document the noise_level parameter in TemplateMatching docstrings and use it when generating noisy stimuli instead of a hard-coded constant.

Tests:
- Add 1D and 2D template matching tests to verify zero-noise behavior, statistical scaling of noise with noise_level, and validation of invalid noise levels.

### CODE REVIEW

File: src/canns/task/tracking.py [hunk 1]
```diff
@@ -297,7 +299,11 @@ def __init__(
             Iext (float | Quantity): The position of the external input.
             duration (float | Quantity): The duration for which the noisy stimulus is presented.
             time_step (float | Quantity, optional): The simulation time step. Defaults to 0.1.
+            noise_level (float, optional): Multiplier on the CANN amplitude ``A`` that
+                scales the per-step Gaussian noise. Must be non-negative. Defaults to 0.1.
         """
+        if noise_level < 0:
+            raise ValueError(f"noise_level must be non-negative, got {noise_level!r}")
```
[the remark this answers is not in the document]
Comment (Routhleck) [reply src/canns/task/tracking.py]: Fixed in 28a0ff0. `noise_level` is now coerced via `float(noise_level)` (TypeError on failure) and then checked with `math.isfinite(...)` (ValueError on `inf` / `nan`). The docstring on both `__init__` and the class is updated to say "finite, non-negative scalar". Tests now also cover `inf`, `nan`, and a non-numeric string input.

**Comment by Routhleck** [comment]
Also addressing the two overall review suggestions in 28a0ff0:

1. **Class-level discoverability of `A` and `noise_level`.** Added an `Attributes:` section to the `TemplateMatching` class docstring listing both `A` and `noise_level` with their roles, so they're surfaced in IDE tooltips / Sphinx output (in addition to the `__init__` docstring and instance assignments). Skipped `__slots__` since the parent `TrackingTask` doesn't use it and adding it would be a wider refactor.

2. **Test floating-point comparison.** Replaced `task.data.std(axis=0).max() == 0.0` with `np.testing.assert_allclose(task.data.std(axis=0), 0.0, atol=1e-12)` in both 1D and 2D `_zero_is_clean` tests.

All 11 tracking tests pass and `uv run pytest -m 'not slow'` is green (139 passed).
